File: packages/plugin/overlay-plugin/src/main/java/com/overlay/plugin/messages/PacketUtils.java
package com.overlay.plugin.messages;

import com.eu.habbo.habbohotel.gameclients.GameClient;
import com.eu.habbo.habbohotel.users.Habbo;
import com.overlay.plugin.messages.outgoing.common.SessionDataComposer;

public class PacketUtils {
  private PacketUtils() {
    throw new IllegalStateException("Utility class");
  }

  public static void updateSessionData(Habbo habbo) {
    if (habbo == null) return;
    GameClient client = habbo.getClient();
    if (client == null) return;

    SessionDataComposer sessionDataComposer = new SessionDataComposer(habbo.getHabboInfo());
    client.sendResponse(sessionDataComposer.compose());
  }
}

File: packages/plugin/overlay-plugin/src/main/java/com/overlay/plugin/messages/PacketUtils.kt
package com.overlay.plugin.messages

import com.eu.habbo.habbohotel.users.Habbo
import com.overlay.plugin.messages.outgoing.common.SessionDataComposer

class PacketUtils {
    companion object {
        @JvmStatic
        fun updateSessionData(habbo: Habbo?) {
            if (habbo == null) return
            val client = habbo.client ?: return

            val sessionDataComposer = SessionDataComposer(habbo.habboInfo)
            client.sendResponse(sessionDataComposer.compose())
        }
    }
}
